src/DataStructures/Tensor/Tensor.hpp

```diff
@@ -597,6 +596,56 @@ struct MakeWithValueImpl<Tensor<double, Structure...>, double> {
   }
 };
 
+template <typename... Structure>
+struct MakeWithValueImpl<Tensor<ComplexDataVector, Structure...>, DataVector> {
+  /// \brief Returns a Tensor whose `ComplexDataVector`s are the same size as
+  /// `input`, with each element set to `value`.
+  static SPECTRE_ALWAYS_INLINE Tensor<ComplexDataVector, Structure...> apply(
+      const DataVector& input, const double value) noexcept {
+    return Tensor<ComplexDataVector, Structure...>(
+        ComplexDataVector{input.size(), std::complex<double>(value, 0.0)});
+  }
+};
+
+template <int Spin, typename... Structure>
+struct MakeWithValueImpl<
+    Tensor<SpinWeighted<ComplexDataVector, Spin>, Structure...>, DataVector> {
+  /// \brief Returns a Tensor whose `ComplexDataVector`s are the same size as
+  /// `input`, with each element set to `value`.
```

Comment:
I think the documentation here should specify that only the real part is set to zero, or is that an assumption to do with the fact that the complex data vectors are 'SpinWeighted'?